quinn-h3/src/proto/connection.rs

```diff
@@ -187,34 +187,36 @@ impl Connection {
         self.pending_streams[ty as usize].reserve(capacity);
     }
 
-    pub fn request_initiated(&mut self, id: StreamId) {
-        if !self.go_away {
-            self.requests_in_flight.insert(id);
-            self.max_id_in_flight = cmp::max(id.0, self.max_id_in_flight);
+    // The remote peer initiated a stream: a request from client or a push from server
+    pub fn remote_stream_initiated(&mut self, id: StreamId) -> Result<()> {
+        match self.go_away {
```

Comment:
I think this might read nicer as an `if let` (although it's unfortunate that the `id.0 > max_id` will need a nested `if`) with early return. Maybe have a look and decide for yourself which you prefer?

Reply:
The `if let` with a nested `if` is more expressive indeed. I wish we could use a combination in `if let` so we wouldn't be tempted to use `match { Pattern if condition => ... } ` syntactic sugar in these situation.

Reply:
I think this is being worked on in the lang team, don't know what the current status is though.